lib/IRGen/GenRecord.h

```diff
@@ -274,9 +274,7 @@ class RecordTypeInfoImpl : public Base,
     // If the fields are not ABI-accessible, use the value witness table.
     if (!AreFieldsABIAccessible) {
       return emitInitializeWithTakeCall(IGF, T, dest, src);
-    }
-
-    if (auto rawLayout = T.getRawLayout()) {
+    } else if (auto rawLayout = T.getRawLayout()) {
```

Comment:
Might want another condition within this one that handles raw layout types that don't `shouldMoveAsLikeType`.

Reply:
That may be a good change for you to make on main.  This PR is aiming to fix a failure to build Synchronization on the 6.0 branch by getting control flow to be the same on release/6.0 as on main.

Reply:
Oh nvm, I didn't see the first condition in this function. I think for raw layout types that aren't `shouldMoveAsLikeType` they are bitwise takable so the memcpy should be fine.